reference/test/solver/multigrid_kernels.cpp

```diff
@@ -1573,146 +1360,4 @@ TYPED_TEST(Multigrid, SolvesStencilSystemByFCycle)
 }
 
 
-TYPED_TEST(Multigrid, SolvesStencilSystemByKfcgCycle)
-{
-    using Mtx = typename TestFixture::Mtx;
-    using value_type = typename TestFixture::value_type;
-
-    auto multigrid_factory =
-        this->get_multigrid_factory(gko::solver::multigrid::cycle::kfcg);
-    auto solver = multigrid_factory->generate(this->mtx);
-    auto b = gko::initialize<Mtx>({-1.0, 3.0, 1.0}, this->exec);
-    auto x = gko::initialize<Mtx>({0.0, 0.0, 0.0}, this->exec);
-
-    solver->apply(b.get(), x.get());
-
-    GKO_ASSERT_MTX_NEAR(x, l({1.0, 3.0, 2.0}), r<value_type>::value);
-}
-
-
-TYPED_TEST(Multigrid, SolvesStencilSystemByKgcrCycle)
-{
-    using Mtx = typename TestFixture::Mtx;
-    using value_type = typename TestFixture::value_type;
-    auto multigrid_factory =
-        this->get_multigrid_factory(gko::solver::multigrid::cycle::kgcr);
-    auto solver = multigrid_factory->generate(this->mtx);
-    auto b = gko::initialize<Mtx>({-1.0, 3.0, 1.0}, this->exec);
-    auto x = gko::initialize<Mtx>({0.0, 0.0, 0.0}, this->exec);
-
-    solver->apply(b.get(), x.get());
-
-    GKO_ASSERT_MTX_NEAR(x, l({1.0, 3.0, 2.0}), r<value_type>::value);
-}
-
-TYPED_TEST(Multigrid, SolvesStencilSystem2ByVCycle)
-{
-    using Mtx = typename TestFixture::Mtx;
-    using value_type = typename TestFixture::value_type;
-    auto multigrid_factory =
-        this->get_mixed_multigrid_factory(gko::solver::multigrid::cycle::v);
-    auto solver = multigrid_factory->generate(this->mtx2);
-    auto mg_level_list = solver->get_mg_level_list();
-
-    solver->apply(this->b2.get(), this->x2.get());
-
-    ASSERT_TRUE((std::dynamic_pointer_cast<
-                 const gko::multigrid::EnableMultigridLevel<value_type>>(
-        mg_level_list.at(0))));
-    ASSERT_TRUE(
-        (std::dynamic_pointer_cast<const gko::multigrid::EnableMultigridLevel<
-             gko::next_precision<value_type>>>(mg_level_list.at(1))));
-    GKO_ASSERT_MTX_NEAR(this->x2, l({1.0, 3.0, 2.0, 1.0, 3.0, 2.0}),
-                        r<value_type>::value);
-}
-
-
-TYPED_TEST(Multigrid, SolvesStencilSystem2ByWCycle)
-{
-    using Mtx = typename TestFixture::Mtx;
-    using value_type = typename TestFixture::value_type;
-    auto multigrid_factory =
-        this->get_mixed_multigrid_factory(gko::solver::multigrid::cycle::w);
-    auto solver = multigrid_factory->generate(this->mtx2);
-    auto mg_level_list = solver->get_mg_level_list();
-
-    solver->apply(this->b2.get(), this->x2.get());
-
-    ASSERT_TRUE((std::dynamic_pointer_cast<
-                 const gko::multigrid::EnableMultigridLevel<value_type>>(
-        mg_level_list.at(0))));
-    ASSERT_TRUE(
-        (std::dynamic_pointer_cast<const gko::multigrid::EnableMultigridLevel<
-             gko::next_precision<value_type>>>(mg_level_list.at(1))));
-    GKO_ASSERT_MTX_NEAR(this->x2, l({1.0, 3.0, 2.0, 1.0, 3.0, 2.0}),
-                        r<value_type>::value);
-}
-
-
-TYPED_TEST(Multigrid, SolvesStencilSystem2ByFCycle)
```

Comment:
F cycles are still present, is this still obsolete?

Reply:
Deleting System2 is mostly due to Mixed type. It will introduce single precision error when mixed_precision is off. It seems to make sense, but I want to check it more.